third_party/move/move-vm/types/src/values/values_impl.rs

```diff
@@ -3710,19 +3744,34 @@ impl<'c, 'l, 'v> serde::Serialize
 
             // Signer.
             (L::Signer, ValueImpl::Container(Container::Struct(r))) => {
```

Comment:
Double checking my understanding: we will always have `signer` type be converted to `Signer` layout. Only at runtime, we will instead treat it as an enum, correct?

Reply:
Correct! The signer can only be constructed at runtime by the vm and we don't store the signer bytes on chain. That's the reason why we can change the representation of it. 

(Any requires types to be not storable, thus it's not possible to store a value that contains signer on chain)